### PR TITLE
[feat] cache allocator warmup for `from_single_model`

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

```
from diffusers import FluxTransformer2DModel
import torch 
import time

ckpt_path = "https://huggingface.co/black-forest-labs/FLUX.1-dev/blob/main/flux1-dev.safetensors"

start = time.perf_counter()
transformer = FluxTransformer2DModel.from_single_file(
    ckpt_path, torch_dtype=torch.bfloat16, device_map="cuda"
)
# transformer = FluxTransformer2DModel.from_single_file(
#     ckpt_path, torch_dtype=torch.bfloat16,
# ).to("cuda")
torch.cuda.synchronize()
end = time.perf_counter()

print(f"Take taken to initialize the model on CUDA: {(end - start)} seconds.")
```

Timing:

```
CUDA placement manual: 4.86 seconds
Device map: 1.09 seconds
```

(On an H100)

### CODE REVIEW

**Comment by sayakpaul** [comment]
@stevhliu it would be awesome to also add a doc entry for this.

@DN6 test needed to be added this PR?

**Comment by stevhliu** [comment]
Added a note about it in https://github.com/huggingface/diffusers/pull/12256 👍

**Comment by sayakpaul** [comment]
Included a test too.
